WordPress/src/main/java/org/wordpress/android/support/SupportHelper.kt

```diff
@@ -0,0 +1,145 @@
+@file:JvmName("SupportHelper")
+
+package org.wordpress.android.support
+
+import android.content.Context
+import android.support.v7.app.AlertDialog
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.GONE
+import android.widget.EditText
+import android.widget.TextView
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.AccountModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.prefs.AppPrefs
+import org.wordpress.android.util.validateEmail
+
+/**
+ * This function will check whether there is a support email saved in the AppPrefs and use the saved email and name
+ * to run the provided function, most likely a support request. If there is no saved support email, it'll trigger a
+ * function to show a dialog with email and name input fields which then it'll save to AppPrefs and run the provided
+ * function.
+ *
+ * @param context Context the dialog will be showed from
+ * @param account WordPress.com account to be used for email and name suggestion in the input dialog
+ * @param selectedSite Selected site to be used for email and name suggestion in case the user is not logged in
+ * @param emailAndNameSelected Function to run with the email and name from AppPrefs or the input dialog
+ */
+fun showSupportIdentityInputDialogAndRunWithEmailAndName(
+    context: Context,
+    account: AccountModel?,
+    selectedSite: SiteModel?,
+    emailAndNameSelected: (String, String) -> Unit
+) {
+    val currentEmail = AppPrefs.getSupportEmail()
+    if (!currentEmail.isNullOrEmpty()) {
+        emailAndNameSelected(currentEmail, AppPrefs.getSupportName())
+    } else {
+        val (emailSuggestion, nameSuggestion) = supportEmailAndNameSuggestion(account, selectedSite)
+        showSupportIdentityInputDialog(context, emailSuggestion, nameSuggestion, false) { email, name ->
+            AppPrefs.setSupportEmail(email)
```

Comment:
I have been thinking about this for some time and I have to say, I am not really sure it's the correct call to update the AppPrefs from here. We could do it from the callers of this function, but that'll mean that we have to update the AppPrefs even if the dialog is not shown OR we have to split the implementation and move some of the logic to the caller.

In the upcoming `HelpActivity` changes, I am doing a similar thing and that also felt wrong, so I thought I'd mention it here.

Reply:
Even though it's descriptive, the name is too long and it becomes harder to read the code, IMHO. Why not just call it `getUserIdentity()`? 

Regarding the AppPrefs update, I don't really have a strong opinion about it and it makes sense to have it here based on your explanation.

Reply:
I have renamed it to `getSupportIdentity` in da3a8a90aca14e00c356d258387ee7663f0734e5. I don't think `getUserIdentity` reflects what it does.

FYI: I expect this method to be completely removed during the refactor.